### DISCUSSION TITLE
When is the ﻿metadata_modification_time updated in system.table?

### DISCUSSION THREAD

**Discussion by calebeaires** [ClickHouse/ClickHouse#63958] [Q&A]
Specifically, I'd like to know in which operations (SELECT, INSERT, UPDATE) the ﻿metadata_modification_time is updated. Can anyone provide insight into this behavior?

Is there a way to force the update of ﻿metadata_modification_time?

**Comment by den-crane** [comment]
metadata_modification_time reflects the modification time of .sql file of the related table.
usually it reflects the time of the last `alter table`.

```
:) select table, metadata_modification_time, from system.tables where table = 'X' limit 10;

 ┌─table─┬─metadata_modification_time─┐
 │ X     │        2023-12-02 21:56:59 │
 └───────┴────────────────────────────┘

# touch /var/lib/clickhouse/metadata/default/X.sql

:) select table, metadata_modification_time, from system.tables where table = 'X' limit 10;
┌─table─┬─metadata_modification_time─┐
│ X     │        2024-05-16 15:05:31 │
└───────┴────────────────────────────┘
```
